### PR TITLE
Sept 2021 alignment

### PR DESCRIPTION
**Pull request by hey-waffles**

### CODE REVIEW

File: src/client/data/controllers/RulesetController.ts [hunk 1]
```diff
@@ -1,8 +1,23 @@
 import { RulesetDocument } from "types/documents";
+import { rest } from "utilities/request";
 import { RulesetManager } from "../managers";
 import { DataController } from "./DataController";
 
 class $RulesetController extends DataController<RulesetDocument> {
+  public async updateIsPublic(id: string, isPublic: boolean) {
```
Comment (hey-waffles) [inline src/client/data/controllers/RulesetController.ts:7]: Comment

File: src/pages/api/users.ts [hunk 2]
```diff
@@ -17,6 +17,13 @@ async function getUsers(this: HTTPHandler, req: NextApiRequest) {
 }
 
 
+async function updateUser(this: HTTPHandler, req: NextApiRequest) {
```
Comment (hey-waffles) [inline src/pages/api/users.ts:20]: Comment

File: src/pages/campaigns/[id]/index.tsx [hunk 3]
```diff
@@ -9,127 +9,167 @@ import { query as q } from "faunadb";
...
-    client.query(
-      q.Call(
-        `create_campaign_invite`,
-        [ props.campaign.id, inviteKey ]
-      )
-    ).then((res) => {
-      console.log("tmp");
+
+  React.useEffect(() => {
+    console.log("first");
```
Comment (hey-waffles) [inline src/pages/campaigns/[id]/index.tsx:117]: Remove log

File: src/pages/campaigns/[id]/index.tsx [hunk 4]
```diff
@@ -9,127 +9,167 @@ import { query as q } from "faunadb";
...
+    UserController.readMissing(playerIDs)
+    .then(() => {
+      const newPlayers = UserManager.getMany(playerIDs);
+      setPlayers(newPlayers);
+    });
+  }, []);
+
+  React.useEffect(() => {
+    const newCampaign = CampaignManager.get(props.campaign.id);
+    console.log(newCampaign);
```
Comment (hey-waffles) [inline src/pages/campaigns/[id]/index.tsx:138]: Remove log :(

File: src/server/logic/ImageLogic.ts [hunk 5]
```diff
@@ -63,17 +72,42 @@ export const ImageLogic = (new FaunaLogicBuilder("images")
   .done()
 .done()).export();
 
+ImageLogic.create = async (image: Partial<ImageDocument>, method: AssetUploadSource, myUser: MyUserDocument) => {
```
Comment (hey-waffles) [inline src/server/logic/ImageLogic.ts:75]: Comment

**Review by lucyawrey — APPROVED** [review]
Looks good, I really like your progress so far on the backend.
<details><summary>Browser metadata</summary>





```
Path:      /
Browser:   Chrome 94.0.4606.61 on Windows 10
Viewport:  1319 x 1142 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-139--reroll.netlify.app%2F&os=Windows&os_version=10&browser=Chrome&browser_version=94.0&resolution=1319x1142&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-139--reroll.netlify.app/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=615a049a7dd2950008bd0a44&commentID=615a18620e0b48cfe0be47ca&resolution=resolved)

**Pull request merged**